python/pyarrow/table.pxi

```diff
@@ -77,6 +77,52 @@ cdef class ChunkedArray:
         self._check_nullptr()
         return self.chunked_array.null_count()
 
+    def __getitem__(self, key):
+        cdef int64_t item
+        cdef int i
+        self._check_nullptr()
+        if isinstance(key, slice):
+            return _normalize_slice(self, key)
+        elif isinstance(key, six.integer_types):
+            item = key
+            if item >= self.chunked_array.length() or item < 0:
+                return IndexError("ChunkedArray selection out of bounds")
```

Comment:
If we allow negative slice bounds, I would expect us to also allow negative indices. Seems like it's time for a `_normalize_index` function?

Reply:
Agreed, perhaps let's handle this as a follow up patch